### PR TITLE
add counter badge rule

### PR DESCRIPTION
**Pull request by yonashailug**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ module.exports = {
         "dialogbody-needs-title-content-and-actions": require("./rules/dialogbody-needs-title-content-and-actions"),
         "dialogsurface-needs-aria": require("./rules/dialogsurface-needs-aria"),
         "spinner-needs-labelling": require("./rules/spinner-needs-labelling"),
-        "badge-needs-accessible-name": require("./rules/badge-needs-accessible-name")
+        "badge-needs-accessible-name": require("./rules/badge-needs-accessible-name"),
+        "counter-badge-needs-count": require("./rules/counter-badge-needs-count")
```
Comment (Harsh-Modi278) [inline lib/index.js:44]: Please mark the component in coverage.md file as well.

File: lib/rules/counter-badge-needs-count.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                                fix(fixer) {
+                                    const ariaLabelFix = fixer.insertTextAfter(iconElement.openingElement.name, ' aria-label=""');
+                                    return ariaLabelFix;
+                                }
+                            });
+                        }
+                    }
+                }
+
+                const hasCount = hasProp(openingElement.attributes, "count");
```
Comment (aubreyquinn) [inline lib/rules/counter-badge-needs-count.js:70]: I think we can move this logic higher up. If there is a count, then the component has an accessible name via text content.

File: tests/lib/rules/counter-badge-needs-count.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+ruleTester.run("counter-badge-needs-count", rule, {
+    valid: [
+        `<CounterBadge count={5} appearance="filled" />`,
+        `<CounterBadge icon={<PasteIcon aria-label="paste" />} count={1} />`,
```
Comment (aubreyquinn) [inline tests/lib/rules/counter-badge-needs-count.js:30]: The CounterBadge component applies aria-hidden to the icon, so it's only for looks and we don't need an aria-label if the count prop is provided.

**Comment by yonashailug** [comment]
> @microsoft-github-policy-service agree [company="Microsoft"]

**Review by aubreyquinn — APPROVED** [review]

**Pull request merged**